tests/rptest/tests/rpk_cluster_test.py

```diff
@@ -184,3 +187,61 @@ def test_cluster_down(self):
             pass
         else:
             assert False, f"Unexpected success: '{r}'"
+
+    @cluster(num_nodes=3)
+    def test_upload_and_query_cluster_license_rpk(self):
+        """
+        Test uploading and retrieval of license via rpk
+        """
+        license = os.environ.get("REDPANDA_SAMPLE_LICENSE", None)
+        if license is None:
+            is_ci = os.environ.get("CI", "false")
+            assert is_ci == "false"
+            self.logger.info(
+                "Skipping test, REDPANDA_SAMPLE_LICENSE env var not found")
+            return
+
+        with tempfile.NamedTemporaryFile() as tf:
+            with open(tf.name, 'w') as f:
+                f.write(license)
+                f.seek(0)
+
+                output = self._rpk.license_set(tf.name)
+                assert "Successfully uploaded license" in output
+
+        def condition():
+            output = self._rpk.license_info()
+
+            expected = '''LICENSE INFORMATION:
```

Comment:
This is nice, just thinking the command could also support a `--json` option which would make testing easier

Reply:
Also I see we are not testing the number of days. I think that should be added to the test, since we know the exact date the license expires, just subtract that date from todays date and compare that to the value observed in `output['expires']`.

Here's an [example](https://github.com/redpanda-data/redpanda/blob/dev/tests/rptest/tests/cluster_features_test.py#L242-L243)

Reply:
Done, thanks!